### ISSUE TITLE
Problem to read a profile from RF627 Smart scanner with Python

### ISSUE THREAD

**Issue by matousc89** [RIFTEK-LLC/RF62X-SDK#7]
I downloaded the folder with prepared libraries and examples. It connects to scanner correctly (as far as I can tell):

```
#########################################
#                                       #
#          Profile Example v2.x.x       #
#                                       #
#########################################
SDK version 2.19.0
=========================================
Search scanners from:
 * IP Address	:  192.168.15.100 
 * Netmask	:  255.255.255.0
Discovered:  1  RF627-Smart
-----------------------------------------
Was found	: 1 RF627 v2.x.x
=========================================


ID scanner's list: 1
-----------------------------------------
Device information:
* Name		:  2D laser scanner
* Serial	:  210200
* IP Addr	:  192.168.15.20
Profile was not received!
-----------------------------------------
```
However, the profile is not received

I modified your code to see at least something about the problem:

```
def py_profile2python(profile_sdk):
    ret={}
    if profile_sdk is not None:
        scanner_type=profile_sdk.contents.type
        if scanner_type==kRF627_OLD:
            try:
                profile=profile_sdk.contents.rf627old_profile2D.contents
                format=rf627_old_point2D_t
            except:
                return None
        elif scanner_type==kRF627_SMART:
            try:
                # FIXME we got stuck here
                profile=profile_sdk.contents.rf627old_profile2D.contents
                profile=profile_sdk.contents.rf627smart_profile2D.contents
                format=rf627_smart_profile2D_t
            except Exception as e:
                print(e)
                return None
```

And that provides me with this information:

```
Traceback (most recent call last):
  File "C:\Users\Swag\Desktop\matous_app\FRAME_EXAMPLE.py", line 44, in <module>
    frame=get_frame(scanner, kSERVICE)
  File "C:\Users\Swag\Desktop\matous_app\PYSDK_SMART.py", line 4042, in get_frame
    frame=py_frame2python(_frame)
  File "C:\Users\Swag\Desktop\matous_app\PYSDK_SMART.py", line 3946, in py_frame2python
    if memmove(_buffer_frame_tmp, v, c*sizeof(c_char)):
OSError: exception: access violation reading 0x00007FFB4FC41000
```

Please advise. What am I doing wrong? What should I try?

**Comment by matousc89** [comment]
After few months I still have not found solution. 

I have a "newer" scanner that works well. In web applications, both scanners have hardware version 18.12.7, altought they are not identical! One has red laser, second one has blue laser. Also the web application looks quite different. The firmware version of newer one is 2.4.0, while the older one has firmware version18.12.7. It is a bit confusing.

When I used the code for reading profiles (working code from newer scanner), I get the error:
> NULL pointer access

In web appliacation, it is possible to read data from the older scanner (it is working). So it seems:

- those scanners are not identical
- I can connect to both of them via Python
- I can read data only from one of them with Python, even though the "unreadable" scanner is not broken
- most likely the problem is in Python wrapper

Please help.

**Comment by KuzVlad** [comment]
Regarding your problem of getting profiles from the old version of the scanner. Last time we tried to repeat your behavior, but we were unable to repeat it. Therefore, we can offer 2 ways to solve the problem:
1. Update your firmware to a newer one (2.х.х)
2. Our specialist will try to connect to you remotely and debug SDK with your scanner based on the old firmware on your computer (unfortunately, we do not have a scanner with the same software version as yours).

So, I would advise you to select the first option and update the scanner to a newer software version (2.x.x), since the production and support of older versions of scanners is limited.

**Comment by matousc89** [comment]
@KuzVlad I have tried to upgrade the firmware (https://cloud.riftek.com/index.php/s/oekgw7LnziE8x2R) via the web interface. It required different filename (extension) that provided. I changed it manually, but I was still unable to upload it. I also one older version (the same that is on the second scanner):
![image](https://github.com/RIFTEK-LLC/RF62X-SDK/assets/18695652/7b23a033-ff86-453c-9297-d0055accd276)

What I am doing wrong? What exact firmware package I should use and how to force the upgrade?
